apps/meteor/client/providers/UserProvider/UserProvider.tsx

```diff
@@ -74,13 +76,22 @@ const UserProvider = ({ children }: UserProviderProps): ReactElement => {
 			querySubscription: createReactiveSubscriptionFactory<ISubscription | undefined>((query, fields, sort) =>
 				Subscriptions.findOne(query, { fields, sort }),
 			),
-			queryRoom: createReactiveSubscriptionFactory<IRoom | undefined>((query, fields) => Rooms.findOne(query, { fields })),
+			queryRoom: createReactiveSubscriptionFactory<IRoom | undefined>((query) => {
+				const predicate = createPredicateFromFilter(query);
+				return Rooms.state.find(predicate);
+			}),
 			querySubscriptions: createReactiveSubscriptionFactory<SubscriptionWithRoom[]>((query, options) => {
 				if (userId) {
 					return Subscriptions.find(query, options).fetch();
 				}
+				// Anonnymous users don't have subscriptions. Fetch Rooms instead.
+				const predicate = createPredicateFromFilter(query);
+				const rooms = Rooms.state.filter(predicate) as unknown as SubscriptionWithRoom[]; // FIXME: this is a hack to make the typings work. How was it working before?
```

Comment:
```suggestion
				const rooms = Rooms.state.filter(predicate);
```
https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks

Since pick is done like this: 
![Screenshot 2025-07-07 at 11 06 22](https://github.com/user-attachments/assets/c047da1d-7ff7-4f56-8fe9-a83dee0a87c2)
No excess-property-checks are done